### PR TITLE
Windows URI association fix

### PR DESCRIPTION
**Pull request by FrayxRulez**

### CODE REVIEW

**Comment by FrayxRulez** [comment]
Everything ok now I think... @auchri

File: Telegram/SourceFiles/pspecific_win.cpp [hunk 1]
```diff
@@ -824,6 +824,16 @@ void RegisterCustomScheme() {
 	if (!_psOpenRegKey(L"Software\\Classes\\tg\\shell\\open", &rkey)) return;
 	if (!_psOpenRegKey(L"Software\\Classes\\tg\\shell\\open\\command", &rkey)) return;
 	if (!_psSetKeyValue(rkey, 0, '"' + exe + qsl("\" -workdir \"") + cWorkingDir() + qsl("\" -- \"%1\""))) return;
+	
+	if (!_psOpenRegKey(L"Software\\TelegramDesktop", &rkey)) return;
+	if (!_psOpenRegKey(L"Software\\TelegramDesktop\\Capabilities", &rkey)) return;
+	if (!_psSetKeyValue(rkey, L"ApplicationName", qls("TelegramDesktop"))) return;
```
Comment (auchri) [inline Telegram/SourceFiles/pspecific_win.cpp]: Shouldn't this be `Telegram Desktop`?

File: Telegram/SourceFiles/pspecific_win.cpp [hunk 2]
```diff
@@ -824,6 +824,16 @@ void RegisterCustomScheme() {
...
+	if (!_psSetKeyValue(rkey, L"ApplicationDescription", qls("Telegram Desktop"))) return;
+	if (!_psOpenRegKey(L"Software\\TelegramDesktop\\Capabilities\\UrlAssociations", &rkey)) return;
+	if (!_psSetKeyValue(rkey, L"tg", qls("tg"))) return;
+	
+	if (!_psOpenRegKey(L"Software\\RegisteredApplications", &rkey)) return;
+	if (!_psSetKeyValue(key, L"TelegramDesktop", qls("SOFTWARE\\TelegramDesktop\\Capabilities"))) return;
```
Comment (auchri) [inline Telegram/SourceFiles/pspecific_win.cpp]: Here too

**Comment by FrayxRulez** [comment]
... The git client I used this time just lost the < > in the commit message... I hope it's not a problem :S

**Comment by auchri** [comment]
> ... The git client I used this time just lost the < > in the commit message... I hope it's not a problem :S

It will be ok for this time 😉

File: Telegram/SourceFiles/pspecific_win.cpp [hunk 3]
```diff
@@ -824,6 +824,16 @@ void RegisterCustomScheme() {
 	if (!_psOpenRegKey(L"Software\\Classes\\tg\\shell\\open", &rkey)) return;
 	if (!_psOpenRegKey(L"Software\\Classes\\tg\\shell\\open\\command", &rkey)) return;
 	if (!_psSetKeyValue(rkey, 0, '"' + exe + qsl("\" -workdir \"") + cWorkingDir() + qsl("\" -- \"%1\""))) return;
+	
+	if (!_psOpenRegKey(L"Software\\TelegramDesktop", &rkey)) return;
+	if (!_psOpenRegKey(L"Software\\TelegramDesktop\\Capabilities", &rkey)) return;
+	if (!_psSetKeyValue(rkey, L"ApplicationName", qls("TelegramDesktop"))) return;
+	if (!_psSetKeyValue(rkey, L"ApplicationDescription", qls("Telegram Desktop"))) return;
+	if (!_psOpenRegKey(L"Software\\TelegramDesktop\\Capabilities\\UrlAssociations", &rkey)) return;
+	if (!_psSetKeyValue(rkey, L"tg", qls("tg"))) return;
```
Comment (auchri) [inline Telegram/SourceFiles/pspecific_win.cpp]: @FrayxRulez ping

Comment (FrayxRulez) [inline Telegram/SourceFiles/pspecific_win.cpp]: Ok, creating a new commit

Comment (FrayxRulez) [inline Telegram/SourceFiles/pspecific_win.cpp]: Done!

**Comment by FrayxRulez** [comment]
I'm talking with @john-preston about what to do, since Default Programs are supported since Windows XP, that is the minimum supported Windows version by tdesktop

**Comment by FrayxRulez** [comment]
Ok, I've tested the code on Windows XP and effectively the "legacy" URI registration is needed. Added with the last commit

**Pull request merged**